src/ck-core/ckfutures.h

```diff
@@ -84,7 +87,7 @@ namespace ck {
       CkSendToFuture(handle_, msg);
     }
 
-    bool is_ready() { return CkProbeFuture(handle_); }
+    bool const is_ready() const { return CkProbeFuture(handle_); }
```

Comment:
This prevents assignment to the result, e.g. `f.is_ready() = false;`

Reply:
I can see that if this were returning a `bool &`, but an expression like that that should get its own warning for assigning to a temporary if the function returns by value.

The LGTM alert introduced is for this line:

> The 'const' modifier has no effect on return types. The 'const' modifying the return type can be removed.

Reply:
It shouldn't be assignable in the first place since it will return an rvalue. Separately, using `const` on the return type in general prevents moving the returned object/value, so it should be avoided when returning by value.

Reply:
same as above, I did not mean to add `const` before `is_ready`, only after.

Reply:
@rbuch the usual case (with or without `const`) is copy elision, i.e., in-place assignment (hence no copy or move constructor invocation). But yes, I should've seen that it returns rvalue so a `const` there does no good.